### PR TITLE
Task to send token expiration reminder

### PR DESCRIPTION
**Pull request by jingcheng16**
## Technical Summary

Send reminder to enterprise admin, if their idp's api secret is going to expire. This email should be sent out if `idip.date_api_secret_expiration` is filled out and `idp.enable_user_deactivation` is set to True on that IdentityProvider.

Tech Spec: https://docs.google.com/document/d/1LOF1lCl3wTsEQkEEsHtZi3u4N6kvjXaLv-Lu6e9aYjo/edit
Ticket: https://dimagi.atlassian.net/browse/SAAS-15234

### Safety story

The worst case is we sent wrong email out...so I think it is kind of safe?

### Automated test coverage

corehq.apps.sso.tests.test_tasks.py

### QA Plan

No QA at this stage

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/sso/templates/sso/email/api_secret_expiring_reminder.html [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+<p>
+  {% blocktrans %}
+    Best regards,
+  {% endblocktrans %}
+</p>
+
+<p>
+  {% blocktrans %}The CommCare HQ Team{% endblocktrans %}<br />
+  {{ base_url }}
+</p>
```
Comment (biyeun) [inline corehq/apps/sso/templates/sso/email/api_secret_expiring_reminder.html:34]:
```suggestion
</p>

```
nit new line at end of file

File: corehq/apps/sso/tasks.py [hunk 2]
```diff
@@ -99,3 +101,58 @@ def send_idp_cert_expires_reminder_emails(num_days):
...
+    queryset = IdentityProvider.objects.filter(
+        is_active=True,
+        enable_user_deactivation=True,
+        date_api_secret_expiration__gte=date_in_n_days,
+        date_api_secret_expiration__lt=day_after_that,
+    )
+    for idp in queryset.all():
+        context = get_api_secret_expiration_email_context(idp)
+        if not context["to"]:
+            log.error(f"no admin email addresses for IdP: {idp}")
```
Comment (biyeun) [inline corehq/apps/sso/tasks.py:135]: maybe this should be a `notify_exception`?

File: corehq/apps/sso/tasks.py [hunk 3]
```diff
@@ -99,3 +101,58 @@ def send_idp_cert_expires_reminder_emails(num_days):
...
+                log.info(
+                    "Sent %(num_days)s-day api secret expiration reminder "
+                    "email for %(idp_name)s to %(send_to)s." % {
+                        "num_days": num_days,
+                        "idp_name": idp.name,
+                        "send_to": send_to,
+                    }
+                )
+        except Exception as exc:
+            log.error(
```
Comment (biyeun) [inline corehq/apps/sso/tasks.py:155]: i feel like this also should be a `notify_exception`. we want to fail loudly around not being able to send these messages

Comment (jingcheng16) [reply corehq/apps/sso/tasks.py]: Sorry, made a mistake and fixed here: dab3e08e78ff875be2dfc501d878e32b93672bbf

File: corehq/apps/sso/templates/sso/email/api_secret_expiring_reminder.html [hunk 4]
```diff
@@ -0,0 +1,34 @@
+{% load i18n %}
+<p>{% blocktrans %}Dear enterprise administrator,{% endblocktrans %}</p>
+
+<p>
+  {% blocktrans %}
+  The api secret for your Identity Provider {{ idp_name }} expires soon!
```
Comment (biyeun) [inline corehq/apps/sso/templates/sso/email/api_secret_expiring_reminder.html:6]:
```suggestion
  The API Secret for your Identity Provider "{{ idp_name }}" expires soon!
```

File: corehq/apps/sso/templates/sso/email/api_secret_expiring_reminder.txt [hunk 5]
```diff
@@ -0,0 +1,15 @@
+{% load i18n %}
+{% blocktrans %}
+Dear enterprise administrator,
+
+The api secret for your Identity Provider {{ idp_name }} expires soon!
```
Comment (biyeun) [inline corehq/apps/sso/templates/sso/email/api_secret_expiring_reminder.txt:5]:
```suggestion
The API Secret for your Identity Provider "{{ idp_name }}" expires soon!
```

File: corehq/apps/sso/templates/sso/email/api_secret_expiring_reminder.html [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  If the secret is not recreated with a new expiration date, the auto-deactivation of web user for this IdP will stop working {{ expires_on }}
```
Comment (biyeun) [inline corehq/apps/sso/templates/sso/email/api_secret_expiring_reminder.html:7]:
```suggestion
  If a new API Secret is not created, the Remote User Management functionality for your Identity Provider will stop working on: {{ expires_on }}
```

File: corehq/apps/sso/templates/sso/email/api_secret_expiring_reminder.txt [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+If the secret is not recreated with a new expiration date, the auto-deactivation of web user for this IdP will stop working {{ expires_on }}
```
Comment (biyeun) [inline corehq/apps/sso/templates/sso/email/api_secret_expiring_reminder.txt:6]:
```suggestion
If a new API Secret is not created, the Remote User Management functionality for your Identity Provider will stop working on: {{ expires_on }}
```

**Review by biyeun — APPROVED** [review]

**Pull request merged**